scripts/zkai_attention_derived_d128_input_gate.py

```
@@ -0,0 +1,648 @@
+    payload: Any,
+    *,
+    expected: dict[str, Any] | None = None,
+    context: dict[str, Any] | None = None,
+) -> None:
+    data = _dict(payload, "payload")
+    context = build_core_context() if context is None else context
+    validate_core_payload(data, expected=expected, context=context)
+    if set(data) == FINAL_KEYS:
+        cases = _list(data.get("cases"), "cases")
+        if data.get("mutation_inventory") != list(EXPECTED_MUTATIONS):
+            raise AttentionDerivedD128InputError("mutation inventory drift")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)